### PR TITLE
Python 3: Fix DHCP6 + PPP/PPTP

### PR DESCRIPTION
**Pull request by gpotter2**

### CODE REVIEW

File: scapy/utils.py [hunk 1]
```diff
@@ -75,7 +75,7 @@ def lhex(x):
 
 def to_hex(x):
     """Turns string or bytes to hex format"""
-    return x if isinstance(x, bytes) and not six.PY2 else x.encode("hex")
+    return bytes_hex(x).decode()
```
Comment (p-l-) [inline scapy/utils.py]: What is the expected difference between `to_hex()` and `bytes_hex()`?

Comment (gpotter2) [reply scapy/utils.py]: to_hex should return hex as string, whereas bytes_hex returns hex as bytes.

Comment (p-l-) [reply scapy/utils.py]: What do you think of using explicitly `bytes_hex(x).decode()` and remove `to_hex()`? I find this a bit confusing.

**Comment by gpotter2** [comment]
Done

File: scapy/layers/ppp.py [hunk 2]
```diff
@@ -581,7 +581,7 @@ def dispatch_hook(cls, _pkt=None, *_, **kargs):
             code = orb(_pkt[0])
         elif "code" in kargs:
             code = kargs["code"]
-            if isinstance(code, basestring):
+            if isinstance(code, str):
```
Comment (p-l-) [inline scapy/layers/ppp.py]: I think `six.string_types` would be more appropriate here.

**Comment by gpotter2** [comment]
Updated. You may cancel all the duplicated travis ci build :)

**Comment by p-l-** [comment]
@gpotter2 are you sure the PR has been updated?

File: scapy/layers/ppp.py (hunk 2 above)
Comment (gpotter2) [reply scapy/layers/ppp.py]: forgot to push...

Comment (p-l-) [reply scapy/layers/ppp.py]: (I was talking about this comment @gpotter2)

Comment (gpotter2) [reply scapy/layers/ppp.py]: (and I mean it, it's not "outdated")

**Pull request merged**